File: src/main/java/io/lsdconsulting/lsd/distributed/repository/InterceptedDocumentRepository.java
package io.lsdconsulting.lsd.distributed.repository;

import io.lsdconsulting.lsd.distributed.model.InterceptedInteraction;

import java.util.List;

public interface InterceptedDocumentRepository {
    void save(final InterceptedInteraction interceptedInteraction);
    List<InterceptedInteraction> findByTraceIds(final String... traceId);
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/connector/repository/InterceptedDocumentRepository.kt
package io.lsdconsulting.lsd.distributed.connector.repository

import io.lsdconsulting.lsd.distributed.connector.model.InterceptedInteraction

interface InterceptedDocumentRepository : BaseInterceptedDocumentRepository {
    fun save(interceptedInteraction: InterceptedInteraction)
    fun findByTraceIds(vararg traceId: String): List<InterceptedInteraction>
}
